### PR TITLE
Remove `copy_dataframe` parameter, update requirements with release of Woodwork v0.0.6

### PR DESCRIPTION
**Pull request by angela97lin**
- Remove use of `copy_dataframe` parameter which was removed in 0.0.6
- Update Woodwork requirement to >=0.0.6

### CODE REVIEW

File: evalml/tests/automl_tests/test_automl.py [hunk 1]
```diff
@@ -1776,7 +1776,7 @@ def test_automl_woodwork_user_types_preserved(mock_binary_fit, mock_binary_score
             assert arg.semantic_tags['cat col'] == {'category'}
             assert arg.logical_types['cat col'] == ww.logical_types.Categorical
             assert arg.semantic_tags['num col'] == {'numeric'}
-            assert arg.logical_types['num col'] == ww.logical_types.WholeNumber
```
Comment (angela97lin) [inline evalml/tests/automl_tests/test_automl.py:1779]: WholeNumber was removed

File: evalml/utils/gen_utils.py [hunk 2]
```diff
@@ -300,9 +300,11 @@ def _convert_to_woodwork_structure(data):
         ww_data = pd.DataFrame(ww_data)
 
     # Convert pandas data structures to Woodwork data structures
+    ww_data = ww_data.copy()
     if isinstance(ww_data, pd.Series):
         return ww.DataColumn(ww_data)
-    return ww.DataTable(ww_data, copy_dataframe=True)
+
+    return ww.DataTable(ww_data)
```
Comment (dsherry) [inline evalml/utils/gen_utils.py:307]: Hmm, so is `DataTable` copying implicitly now? The [docs for the constructor](https://woodwork.alteryx.com/en/stable/generated/woodwork.datatable.DataTable.html#woodwork.datatable.DataTable) just say "create datatable"

**Review by freddyaboulton — APPROVED** [review]
@angela97lin Thanks for the fix!

**Review by dsherry — APPROVED** [review]
Thanks @angela97lin !

One question: why didn't our dependency update bot catch this? Shouldn't we have one or more unit tests which should have failed with the removal of `WholeNumber` and/or `copy_dataframe`?

**Review by bchen1116 — APPROVED** [review]
LGTM!

File: evalml/utils/gen_utils.py (hunk 2 above)
Comment (angela97lin) [reply evalml/utils/gen_utils.py:307]: Yup, see description for https://github.com/Featurelabs/woodwork/pull/398!

Comment (dsherry) [reply evalml/utils/gen_utils.py:307]: Thanks @angela97lin ! Filed https://github.com/FeatureLabs/woodwork/issues/420

**Review by jeremyliweishih — APPROVED** [review]
very cool!

**Comment by angela97lin** [comment]
@dsherry Not sure why our dependency bot didn't catch this... but our unit tests didn't fail because the last merge happened before woodwork released. Just reran tests on main and they break, as expected: https://app.circleci.com/pipelines/github/alteryx/evalml/7581/workflows/c7130efe-954f-4c2a-a8eb-5e29a90eeedc/jobs/85802

**Comment by dsherry** [comment]
@angela97lin oh I know why... we don't have `woodwork` on the allow-list for the deps update bot! Fixing that now :)

**Comment by angela97lin** [comment]
@dsherry Ah, for some reason I thought the list was directly taken from `core-requirements.txt` 🤦‍♀️ Thanks!

**Pull request merged**